src/Infrastructure/Util/src/ESMF_LogErr.F90

```diff
@@ -1730,11 +1730,80 @@ subroutine ESMF_LogInitialize(filename, logappendflag, logkindflag, rc)
 !      \end{description}
 !
 !EOPI
+
+    integer                        :: localrc
+    integer                        :: msgAbortCnt
+    type(ESMF_LogMsg_Flag)         :: msgAbortLst(6)
+    character(ESMF_MAXSTR)         :: envRtAbort
+    integer                        :: idxRtAbort
+    type(ESMF_LogPrivate), pointer :: alog
+
     ! Initialize return code; assume routine not implemented
     if (present(rc)) rc = ESMF_RC_NOT_IMPL
 
     call ESMF_LogOpen(ESMF_LogDefault, filename,  &
-        appendflag=logappendflag, logkindflag=logkindflag, rc=rc)
+        appendflag=logappendflag, logkindflag=logkindflag, rc=localrc)
+    if (localrc /= ESMF_SUCCESS) then
+      write (ESMF_UtilIOStderr,*) ESMF_METHOD,  &
+          ": ESMF_LogOpen - ESMF_LogDefault"
+      call ESMF_UtilIOUnitFlush(ESMF_UtilIOStderr, rc=rc)
+      if (present(rc)) rc=localrc
+      return
+    end if
+
+    alog => ESMF_LogTable(ESMF_LogDefault%logTableIndex)
+    call c_ESMC_VMGetEnv("ESMF_RUNTIME_LOGMSGABORT", envRtAbort, localrc)
```

Comment:
Too bad this file in `Util` is compiled _before_ `ESMF_VMGetEnv()` is available. Using the `c_ESMC_` interface, as you have it, is the ugly but correct solution.